src/DynamoCore/Scheduler/UpdateGraphAsyncTask.cs

```diff
@@ -77,7 +77,12 @@ internal bool Initialize(EngineController controller, WorkspaceModel workspace)
                 {
                     var node = workspace.Nodes.FirstOrDefault(n => n.GUID.Equals(nodeGuid));
                     if (node != null)
+                    {
                         node.ClearDirtyFlag();
+
+                        // Set IsEvaluating to true, so that node's preview bubble won't crash.
+                        node.IsEvaluating = true;
```

Comment:
OK. So the comment says, it is dangerous to access mirror data outside SchedulerThread, I think that's how crash happens. Does `RunOnSchedulerSync` really runs callback on scheduler thread? Otherwise, if `UpdateGraphAsyncTask` hasn't finished, scheduler won't get a change to pull data from cached value.

So should putting the preview refreshing job in scheduler fix the problem?